In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df_sachin = pd.read_csv("Sachin_ODI.csv")

In [3]:
df_sachin.shape

(360, 14)

In [4]:
df_sachin.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

## What is the probability of winning

In [5]:
df_sachin["Won"].value_counts()

True     184
False    176
Name: Won, dtype: int64

In [6]:
prob_winning = 184/(184 + 176)
prob_winning

0.5111111111111111

In [7]:
df_won = df_sachin[df_sachin["Won"]==True]

In [8]:
len(df_won)

184

In [9]:
prob_winning = len(df_won) / len(df_sachin)
prob_winning

0.5111111111111111

## What is the probability of century

In [10]:
df_sachin["century"].value_counts()

False    314
True      46
Name: century, dtype: int64

In [11]:
prob_century = 46 / (314 + 46)
prob_century

0.12777777777777777

In [12]:
df_century = df_sachin[df_sachin["century"]==True]

In [13]:
len(df_century)

46

In [14]:
prob_century = len(df_century) / len(df_sachin)
prob_century

0.12777777777777777

## Winning and century

In [15]:
df_sachin[["century", "Won"]].value_counts()

century  Won  
False    False    160
         True     154
True     True      30
         False     16
dtype: int64

In [16]:
df_sachin[["century", "Won"]].value_counts().unstack()

Won      False  True 
century              
False      160    154
True        16     30

In [17]:
# Contingency table
pd.crosstab(index=df_sachin["century"], columns=df_sachin["Won"], margins=True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

What is the probability of winning AND century

1) In how many matches has winning and century happened? 30
2) Totally how many matches are there? 360
3) What is the probability of winning and century? 30/360

In [18]:
df_win_and_century = df_sachin[(df_sachin["century"] == True) & (df_sachin["Won"] == True)]

In [19]:
len(df_win_and_century)

30

In [20]:
prob_win_and_century = len(df_win_and_century) / len(df_sachin)
prob_win_and_century

0.08333333333333333

In [21]:
30/360

0.08333333333333333

## Winning OR century

In [22]:
pd.crosstab(index=df_sachin["century"], columns=df_sachin["Won"], margins=True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

What is the probability of winning OR century

1) In how many matches will winning or century happen?
a) 184 + 46
b) 184 + 16
c) 154 + 16
d) 184 + 46 - 30

In [23]:
184 + 46 - 30

200

In [24]:
df_win_or_century = df_sachin[(df_sachin["century"] == True) | (df_sachin["Won"] == True)]

In [25]:
len(df_win_or_century)

200

In [26]:
prob_win_or_century = len(df_win_or_century) / len(df_sachin)
prob_win_or_century

0.5555555555555556

In [27]:
200/360

0.5555555555555556

In [28]:
prob_winning + prob_century - prob_win_and_century

0.5555555555555555

## Complement

In [29]:
# Win and loss
df_won = df_sachin[df_sachin["Won"]==True]
prob_won = len(df_won) / len(df_sachin)
prob_won

0.5111111111111111

In [30]:
df_lost = df_sachin[df_sachin["Won"]==False]
prob_lost = len(df_lost) / len(df_sachin)
prob_lost

0.4888888888888889

In [31]:
prob_won + prob_lost

1.0

In [32]:
# Atleast 8 fours are scored
# Complement: At most 7 fours

In [33]:
df_at_least_eight_fours = df_sachin[df_sachin["fours"] >= 8]
prob_at_least_eight_fours = len(df_at_least_eight_fours) / len(df_sachin)
prob_at_least_eight_fours

0.26666666666666666

In [34]:
len(df_at_least_eight_fours)

96

In [35]:
df_at_most_seven_fours = df_sachin[df_sachin["fours"] <= 7]
prob_at_most_seven_fours = len(df_at_most_seven_fours) / len(df_sachin)
prob_at_most_seven_fours

0.7333333333333333

In [36]:
len(df_at_most_seven_fours)

264

In [37]:
prob_at_least_eight_fours + prob_at_most_seven_fours

1.0